content/software/ide-v1/tutorials/ide-v1-security/ide-v1-security.md

```diff
@@ -0,0 +1,58 @@
+
+- **Apply Defense in Depth**: Layered security mechanisms are in place to increase security as a whole.
+- **Use a Positive Security Model**: A ‘positive’ security model defines what is allowed and rejects everything else.
+- **Fail Securely**: It is important that failures are handled so that exceptions do not enable unwanted behavior.
+- **Run with Least Privileges**: The principle of least privilege is required to perform every business process.
+- **Avoid Security through Obscurity**: Security through obscurity alone is a weak security mechanism, however when combined with all principles it can be used as an additional layer of security.
+- **Keep Security Simple**: Keeping the application’s security simple is a better option than having complex designs.
+- **Assuming compromise**: The assuming compromise principle is useful to improve the detection and response capabilities in order to predict and remediate the security events before they evolve into security incidents.
+- **Keep people away from data**: Usage of mechanisms, patterns and tools to reduce or eliminate the need for direct data access or manual processing data with the aim of reducing the risk of mishandling or modification and human error when handling sensitive data.
+
+Moreover, the members of Arduino take in considerations the following pillars as part of the Arduino Secure Software Development Lifecycle:
+
+- **Education and Guidance**: in order to ensure that developers have specific know-how and receive training on secure architectural and coding standards.
+- **Secure Data Management**: in order to ensure that environments, sensitive data, PII are managed properly and in accordance with legislation, adopting and implementing in a correct way, the necessary security standards.
+- **Secure Repository Management**: with the purpose of guaranteeing that the Least of privilege and Separation of duties principles are in place to protect the software repositories.
+- **Secure Environment**: in order to ensure the adequate segregation of the services and data contained into Development / Staging / Production environments.
+- **Secure Development**: with the purpose of guaranteeing that security risks, security requirements and best practices are taken into consideration in all the steps of the development lifecycle for Hardware, Software and Infrastructure artifacts and reducing the residual risk associated with security flaws and bugs managed through the internal Vulnerability Management Program.
+- **Security Testing**: in order to ensure that all required security activities such as, but not limited to: Design Review, Threat and Risk Analysis, Code Review, Penetration Test, Vulnerability Assessment and all fixing activities related to the Vulnerability Management Program are performed by the security team, with the help of the developers, along all stages of the development lifecycle.

```

Comment:
@rhaidiz if possible, divide up these sentences as they are quite long (around 20 words per sentence is ideal).

Reply:
I very much agree with you, all these sentences should be reworked, but I ultimately decided to keep them like this to be consistent with what we have posted for the Arduino [Cloud Security Considerations](https://docs.arduino.cc/arduino-cloud/features/security-considerations). However, I would be more than happy to rework the sentences both here and in the Arduino Cloud Security article. If @stevisco also agrees I'll do that here and finally I'll align the other document accordingly.